### ISSUE TITLE
Bugs in range method in RTree

### ISSUE THREAD

**Issue by ChenZhongPu** [InitialDLab/Simba#27]
There are bugs in the [range](https://github.com/InitialDLab/Simba/blob/develop/engine/sql/core/src/main/scala/org/apache/spark/sql/index/RTree.scala#L90) method for limited level searching.

You use `queue`, but never `dequeue` from it (just `front`). And when `cur_level == level_limit`, we should also `enqueue` the internal node.

A fixing:

- replace all `front` with `dequeue`
- when `cur_level == level_limit`,

```scala
else if (cur_level == level_limit) {
          estimate += cur_node.m_mbr.calcRatio(query) * cur_node.size
          cur_node.m_child.foreach {
            case RTreeInternalEntry(mbr, node) =>
              if (query.intersects(mbr)) q.enqueue((node, cur_level + 1))
          }
```

**Comment by Skyprophet** [comment]
Agreed. I haven't test by the time I merge it. You have fixed that? I can fix it if you didn't,

**Comment by ChenZhongPu** [comment]
I have fixed.

**Comment by Skyprophet** [comment]
Fixed in PR #23.
